### PR TITLE
tests: work-around hooks running at restart

### PR DESCRIPTION
**Pull request by valentindavid**
Since a new prepare-device hook has been added to the pc gadget, this has been keeping core-base tests from succeeding. This is the work-around we need.

CI was run on https://github.com/snapcore/core-base/pull/107

### CODE REVIEW

File: tests/lib/prepare.sh [hunk 1]
```diff
@@ -261,7 +261,13 @@ EOF
     # the service setting may have changed in the service so we need
     # to ensure snapd is reloaded
     systemctl daemon-reload
+
+    # Leave some time for snapd to finish processing hooks
+    sleep 5
```
Comment (Meulengracht) [inline tests/lib/prepare.sh:266]: Really awesome find!! I just have a question, will 5 seconds *always* be enough? Or is there something else we could poll for? (Just thinking in terms of robustness)

Comment (valentindavid) [reply tests/lib/prepare.sh:266]: Testing in https://github.com/snapcore/core-base/pull/108

Comment (valentindavid) [reply tests/lib/prepare.sh]: It worked. Done.

File: tests/lib/prepare.sh [hunk 2]
```diff
@@ -214,6 +214,15 @@ update_core_snap_for_classic_reexec() {
     done
 }
 
+flush_changes() {
+    sleep 1
+    local timeout=20
+    while [ "${timeout}" -gt 0  ] && snap changes | grep -q Doing; do
```
Comment (mvo5) [inline tests/lib/prepare.sh:220]: Silly question, could this be just: `retry -n 20 sh -c 'snap changes | grep -q Doing'` ?

Comment (valentindavid) [reply tests/lib/prepare.sh]: I am running the CI again. I will squash if that worked.

File: tests/lib/prepare.sh [hunk 3]
```diff
@@ -214,6 +214,11 @@ update_core_snap_for_classic_reexec() {
     done
 }
 
+flush_changes() {
+    sleep 1
```
Comment (Meulengracht) [inline tests/lib/prepare.sh:218]: Why the initial 'sleep 1'? Maybe a comment why it's neccessary, otherwise this looks really good to me!

Comment (valentindavid) [reply tests/lib/prepare.sh:218]: To avoid races where a task is being queued but not yet in `snap changes`.

Comment (valentindavid) [reply tests/lib/prepare.sh:218]: I will add a comment when I get the CI to run again. "retry" seems to have broken it.

Comment (valentindavid) [reply tests/lib/prepare.sh:219]: Done

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks!

**Review by Meulengracht — APPROVED** [review]
Thanks for all the changes!

File: tests/lib/prepare.sh (hunk 2 above)
Comment (valentindavid) [reply tests/lib/prepare.sh]: Done.

**Pull request merged**